src/macports1.0/macports.tcl

```diff
@@ -1148,7 +1148,12 @@ match macports.conf.default."
                 }
             }
         } else {
-            set macports::build_arch {}
+            # List of currently supported CPU architectures
+            if {$tcl_platform(machine) in [list arm64 x86_64 i386 ppc]} {
```

Comment:
Are we sure these are the names that will be used on non-Darwin platforms? Tcl gets this value by calling `uname(3)`. I'm pretty sure I remember `i686` being used on Linux sometimes, for example.

Reply:
Yep this can be improved by pretty much copying https://github.com/macports/macports-base/pull/268. I will extend these checks further in that PR.